### PR TITLE
pythonPackages.singledispatch: move expression

### PR DESCRIPTION
**Pull request by costrouc**
###### Motivation for this change

no change in version just moving singledispatch to python-modules

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by costrouc** [comment]
Need to add myself as maintainer before merging

**Comment by costrouc** [comment]
Requires myself to be added to maintainers (pull request #44655) list before this pull request

**Comment by xeji** [comment]
@GrahamcOfBorg eval

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please squash the commits and correct the commit message

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.singledispatch python3.pkgs.singledispatch

**Comment by dotlambda** [comment]
> Ran 0 tests in 0.000s

Please provide an appropriate `checkPhase` or set `doCheck = false` and add a comment on why you're doing that.

**Comment by costrouc** [comment]
@GrahamcOfBorg build python2.pkgs.singledispatch python3.pkgs.singledispatch

**Comment by costrouc** [comment]
Commits have been squashed and appropriate commit message is given

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.singledispatch python3.pkgs.singledispatch

**Pull request merged**
